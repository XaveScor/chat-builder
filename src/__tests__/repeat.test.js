import * as React from 'react';
import {createPage, questionBubble, input, ConformsForm, Stop} from '..'
import * as renderer from 'react-test-renderer'

it('repeat', done => {
    const page = createPage()
    let count = 0
    page.use(() => {
        if (count === 2) {
            return {
                steps: [],
                nextPage: Stop,
            }
        }
        ++count;
        return {
            steps: [
                {
                    question: questionBubble,
                    questionProps: {
                        question: 'question',
                    },
                    input,
                }
            ],
            nextPage: page,
        }
    })
    let tree
    renderer.act(() => {
        tree = renderer.create(<ConformsForm page={page} />)
    })
        
    setTimeout(() => {
        const res = tree.toJSON()
        expect(res).toMatchSnapshot()
        done()
    }, 1000)
});