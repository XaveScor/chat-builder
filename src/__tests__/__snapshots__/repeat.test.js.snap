// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repeat 1`] = `
Array [
  <p
    className="robot"
  >
    question
  </p>,
  <p
    className="robot"
  >
    question
  </p>,
]
`;
